rx_signal_processing/dsp.hpp

```diff
@@ -63,47 +79,81 @@ class DSPCore {
     void send_ack();
     void send_timing();
 
-
+//TODO(keith): May remove sizes as member variables.
  private:
+
+    //! CUDA stream the work will be associated with.
     cudaStream_t stream;
 
+    //! Sequence number used to identify and acknowledge a pulse sequence.
     uint32_t sequence_num;
+
+    //! Pointer to the socket used to acknowledge the RF samples have been copied to device.
     zmq::socket_t *ack_socket;
+
+    //! Pointer to the socket used to report the timing of GPU kernels.
     zmq::socket_t *timing_socket;
 
+    //! Stores the total GPU process timing once all the work is done.
     float total_process_timing_ms;
+
+
+
+    //! Stores the decimation timing.
     float decimate_kernel_timing_ms;
 
-    cuComplex *rf_samples;
+    //! Pointer to the RF samples on device.
+    cuComplex *rf_samples_d;
     size_t rf_samples_size;
 
-    cuComplex *first_stage_bp_filters;
+    //! Pointer to the first stage bandpass filters on device.
+    cuComplex *first_stage_bp_filters_d;
     size_t first_stage_bp_filters_size;
 
-    cuComplex *second_stage_filters;
-    size_t second_stage_filters_size;
+    //! Pointer to the second stage filters on device.
+    cuComplex *second_stage_filter_d;
+    size_t second_stage_filter_size;
 
-    cuComplex *third_stage_filters;
-    size_t third_stage_filters_size;
+    //! Pointer to the third stage filters on device.
+    cuComplex *third_stage_filter_d;
+    size_t third_stage_filter_size;
 
-    cuComplex *first_stage_output;
+    //! Pointer to the output of the first stage decimation on device.
+    cuComplex *first_stage_output_d;
     size_t first_stage_output_size;
 
-    cuComplex *second_stage_output;
+    //! Pointer to the output of the second stage decimation on device.
+    cuComplex *second_stage_output_d;
     size_t second_stage_output_size;
 
-    cuComplex *third_stage_output;
+    //! Pointer to the output of the third stage decimation on device.
+    cuComplex *third_stage_output_d;
     size_t third_stage_output_size;
 
-    cuComplex *host_output;
+    //! Pointer to the host output samples.
+    cuComplex *host_output_h;
     size_t host_output_size;
 
-    cudaEvent_t initial_start, kernel_start, stop;
+    //! CUDA event to timestamp when the GPU processing begins.
+    cudaEvent_t initial_start;
+
+    //! CUDA event to timestamp when the kernels begin executing.
+    cudaEvent_t kernel_start;
+
+    //! CUDA event to timestamp when the GPU processing stops.
+    cudaEvent_t stop;
+
+    //! Cuda event to timestamp the transfer of RF samples to the GPU.
+    cudaEvent_t mem_transfer_end;
+
+    //! Stores the memory transfer timing.
+    float mem_time_ms;
 
-    SharedMemoryHandler *shr_mem;
+    //! A shared memory handler object that contains RF samples from the USRP driver.
+    SharedMemoryHandler shr_mem;
 
-    void callbackFunc();
 
 };
 
+void postprocess(DSPCore *dp);
```

Comment:
This never was part of the dspcore class. It could be, but its not invoked as as object method. Its spawned off as a seperate thread which means that it would have to be used as a class method. Invoking as a class method or just regular function is not going to change the functionality in this case.